### PR TITLE
New restore subscription android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/38085

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Please reach out to @deeppandya for more details on testing

### CODE REVIEW

**Review by samartnik — APPROVED** [review]
lgtm

File: android/java/org/chromium/chrome/browser/brave_leo/BraveLeoPlansActivity.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import org.chromium.chrome.browser.init.AsyncInitializationActivity;
 import org.chromium.chrome.browser.profiles.ProfileProvider;
 import org.chromium.chrome.browser.util.LiveDataUtil;
+import org.chromium.chrome.browser.vpn.utils.BraveVpnUtils;
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/brave_leo/BraveLeoPlansActivity.java:24]: I would create a common class side by side with `org.chromium.chrome.browser.billing.InAppPurchaseWrapper`(as it's basically related to purchases) that will contain that `startActivityBraveAccount` method. It looks odd that we cross include VpnUtils class in Leo code.

File: android/java/res/values/brave_colors.xml [hunk 2]
```diff
@@ -393,4 +393,10 @@
     <color name="text_secondary">#3f4e55</color>
     <color name="primary_20">#d5dcff</color>
     <color name="button_background">#3F39E8</color>
+    <color name="primitive_primary_20">#d5dcff</color>
+    <color name="primitive_primary_30">#b7c6ff</color>
+    <color name="primitive_primary_70">#3b08c2</color>
+    <color name="primitive_primary_80">#2a118e</color>
```
Comment (SergeyZhukovsky) [inline android/java/res/values/brave_colors.xml:399]: do we have same colors in both light and dark modes?

File: browser/ui/android/strings/android_brave_strings.grd [hunk 3]
```diff
@@ -2311,7 +2311,13 @@ Are you sure you want to do this?
       <ph name="YEARLY_AMOUNT">%1$s</ph> / yr.
     </message>
     <message name="IDS_BRAVE_VPN_RENEWAL_TEXT" desc="Brave VPN plan text">
-      Try free for 7 days. After 7 days, you will be charged the plan price. All subscriptions are auto-renewed but can be cancelled before renewal.
+      Subscriptions will be charged via your iTunes account. Any unused portion of the free trial, if offered, is forfeited when you buy a subscription.\n\nYour subscription will renew automatically unless it is cancelled at least 24 hours before the end of the current period.\n\nYou can manage your subscriptions in Settings.\nBy using Brave, you agree to the Terms of Use and Privacy Policy.
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:2314]: I don't get the `iTunes account` part? Does that text supposed to be used on iOS?

**Comment by SergeyZhukovsky** [comment]
Just a general question: how does it suppose to look when a user has mobile only purchase? Do we always show the restore button? As I don't think we can restore credentials if the account isn't linked.

File: browser/ui/android/strings/android_brave_strings.grd [hunk 4]
```diff
@@ -2311,7 +2311,13 @@ Are you sure you want to do this?
...
+    </message>
+    <message name="IDS_ALREADY_PURCHASED" desc="Brave VPN plan page already purchased text">
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:2316]: `Brave VPN plan page already purchased text` --> `Brave subscription plan page already purchased text`

File: browser/ui/android/strings/android_brave_strings.grd [hunk 5]
```diff
@@ -2311,7 +2311,13 @@ Are you sure you want to do this?
...
+      Already purchased on brave.com?
+    </message>
+    <message name="IDS_REFRESH_YOUR_CREDENTIALS" desc="Brave VPN plan page refresh your credentials action text">
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:2319]: `Brave VPN plan page refresh your credentials action text` --> `Brave subscription plan page refresh your credentials action text`

File: android/java/org/chromium/chrome/browser/brave_leo/BraveLeoPlansActivity.java [hunk 6]
```diff
@@ -48,6 +49,13 @@ protected void triggerLayoutInflation() {
         mMonthlySubscriptionAmountText = findViewById(R.id.monthly_subscription_amount_text);
         mUpgradeButton = findViewById(R.id.tv_upgrade_now);
 
+        TextView refreshCredentialsButton = findViewById(R.id.refresh_credentials_button);
+        refreshCredentialsButton.setOnClickListener(
+                v -> {
+                    BraveVpnUtils.startActivityBraveAccount(
+                            BraveLeoPlansActivity.this, BraveVpnUtils.getBraveAccountRecoverUrl());
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/brave_leo/BraveLeoPlansActivity.java:56]: does it use `product=vpn` in the URL for Leo as well? It doesn't seem right to me.

File: android/java/res/values/brave_colors.xml (hunk 2 above)
Comment (deeppandya) [reply android/java/res/values/brave_colors.xml:399]: we user the same colors for light/dark mode. we don't have specific dark colors for primitive colors.

File: android/java/org/chromium/chrome/browser/settings/BraveLeoPreferences.java [hunk 7]
```diff
@@ -144,12 +148,4 @@ public void onResume() {
         setModel();
         checkLinkPurchase();
     }
-
-    private String getLinkURL() {
-        if (BraveQAPreferences.isLeoStagingUsed()) {
-            return LINK_SUBSCRIPTION_URL_STAGING;
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/settings/BraveLeoPreferences.java:150]: we can delete `LINK_SUBSCRIPTION_URL_STAGING` and `LINK_SUBSCRIPTION_URL`

Comment (deeppandya) [reply android/java/org/chromium/chrome/browser/settings/BraveLeoPreferences.java:150]: oh i am sorry i missed it.

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Pull request merged**